EIPS/eip-3155.md

```diff
@@ -0,0 +1,241 @@
+---
+eip: 3155
+title: EVM trace specification
+author: Martin Holst Swende (@holiman), Marius van der Wijden (@MariusVanDerWijden)
+discussions-to: https://ethereum-magicians.org/t/eip-3155-create-evm-trace-specification/5007
+status: Draft
+type: Standards Track
+category (*only required for Standard Track): Interface
+created: 2020-12-07
+---
+
+<!--You can leave these HTML comments in your merged EIP and delete the visible duplicate text guides, they will not appear and may be helpful to refer to if you edit it again. This is the suggested template for new EIPs. Note that an EIP number will be assigned by an editor. When opening a pull request to submit your EIP, please use an abbreviated title in the filename, `eip-draft_title_abbrev.md`. The title should be 44 characters or less.-->
+
+## Simple Summary
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the EIP.-->
+This EIP introduces a new JSON standard for EVM traces. There has been a de-facto standard which is used by go-ethereum, OpenEthereum, Besu and Nethermind. This document shall document this standard as well as provide stricter guidelines to make new implementations easier.
+
+## Motivation
+<!--The motivation is critical for EIPs that want to change the Ethereum protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the EIP solves. EIP submissions without sufficient motivation may be rejected outright.-->
+The Ethereum Virtual Machine executes all smart contract code on ethereum. In order to easier debug smart contracts and state tests, a de-facto standard was introduced to log every execution step of the EVM. This standard was implemented by go-ethereum, parity and nethermind and Besu. Since the de-facto standard was not well defined, the implementations differed slightly making it hard to develop adequate tooling.
```

Comment:
```suggestion
The Ethereum Virtual Machine executes all smart contract code on ethereum. In order to more easily debug smart contracts and state tests, a de-facto standard was introduced to log every execution step of the EVM. This standard was implemented by go-ethereum, parity and nethermind and Besu. Since the de-facto standard was not well defined, the implementations differed slightly making it hard to develop adequate tooling.
```